infra/partner-mainnet/main.tf

```diff
@@ -0,0 +1,244 @@
+        value = data.google_secret_manager_secret_version.account_sk_secret_id[count.index].secret_data
+      },
+      {
+        name  = "MPC_RECOVERY_CIPHER_SK"
+        value = data.google_secret_manager_secret_version.cipher_sk_secret_id[count.index].secret_data
+      },
+      {
+        name  = "MPC_RECOVERY_SIGN_SK"
+        value = data.google_secret_manager_secret_version.sign_sk_secret_id[count.index] != null ? data.google_secret_manager_secret_version.sign_sk_secret_id[count.index].secret_data : data.google_secret_manager_secret_version.account_sk_secret_id[count.index].secret_data
+      },
+      {
+        name  = "AWS_ACCESS_KEY_ID"
+        value = data.google_secret_manager_secret_version.aws_access_key_secret_id.secret_data
+      },
+      {
+        name  = "AWS_SECRET_ACCESS_KEY"
+        value = data.google_secret_manager_secret_version.aws_secret_key_secret_id.secret_data
+      },
+      {
+        name  = "MPC_RECOVERY_LOCAL_ADDRESS"
+        value = "https://${var.node_configs[count.index].domain}"
+      },
+      {
+        name = "MPC_RECOVERY_SK_SHARE_SECRET_ID"
+        value = var.node_configs["${count.index}"].sk_share_secret_id
+      },
+      {
+        name = "MPC_RECOVERY_ENV",
+        value = var.env
+      }
+    ])
+  }
+}
+
+resource "google_service_account" "service_account" {
+  account_id   = "multichain-partner-${var.env}"
+  display_name = "Multichain ${var.env} Account"
+}
+
+resource "google_project_iam_binding" "sa-roles" {
+  for_each = toset([
+      "roles/datastore.user",
+      "roles/secretmanager.admin",
+      "roles/storage.objectAdmin",
+      "roles/iam.serviceAccountAdmin",
+  ])
+
+  role = each.key
+  members = [
+    "serviceAccount:${google_service_account.service_account.email}"
+   ]
+   project = var.project_id
+}
+
+resource "google_compute_global_address" "external_ips" {
+  count        = length(var.node_configs)
+  name         = "multichain-partner-mainnet-${count.index}"
+  address_type = "EXTERNAL"
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+resource "google_compute_managed_ssl_certificate" "mainnet_ssl" {
+  count = length(var.node_configs)
+  name = "multichain-partner-mainnet-ssl-${count.index}"
+
+  managed {
+    domains = [var.node_configs[count.index].domain]
+  }
+}
+
+module "ig_template" {
+  count      = length(var.node_configs)
+  source     = "../modules/mig_template"
+  network    = var.network
+  subnetwork = var.subnetwork
+  region     = var.region
+  service_account = {
+    email  = google_service_account.service_account.email,
+    scopes = ["cloud-platform"]
+  }
+  name_prefix          = "multichain-partner-mainnet-${count.index}"
+  source_image_family  = "cos-stable"
+  source_image_project = "cos-cloud"
+  machine_type         = "n2d-standard-2"
+
+  startup_script = "docker rm watchtower ; docker run -d --name watchtower -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --debug --interval 3600"
```

Comment:
let's not use the 1hr interval from testnet. Let's make it 1 min or 30s so it's easier for us to push patches.